sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -185,6 +185,8 @@ class RocksDB(
         val latestSnapshotVersion = fileManager.getLatestSnapshotVersion(version)
         val metadata = fileManager.loadCheckpointFromDfs(latestSnapshotVersion, workingDir)
         loadedVersion = latestSnapshotVersion
+        // Initialize maxVersion upon successful load from DFS
+        fileManager.initializeMaxVersion()
```

Comment:
Always do a listing to discover the max version in load() is a waste of resource, you can simply call
_fileManager.setMaxVersion(version)_
Because usually the latest version is loaded.

Reply:
Listing is only done if maxVersion is not initialized based on the base condition of the call. If query loads a previous version, maxVersion will not be the max in the directory when using setMaxVersion. This could make maxVersion value inconsistent with the actual maximum version present, is that correct?

Reply:
setMaxVersion will handle the case where the query load a previous version locally.
maxVersion is an approximate anyway, it doesn't need to be accurate.